publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java

```diff
@@ -106,25 +112,47 @@ protected PublicationResponse processInput(CreatePublicationRequest input, Reque
 
         var customerAwareUserContext = getCustomerAwareUserContextFromLoginInformation(requestInfo);
 
-        var backendClientCredentials = secretsReader.fetchClassSecret(
-            environment.readEnv("BACKEND_CLIENT_SECRET_NAME"),
-            BackendClientCredentials.class);
-        var cognitoServerUri = URI.create(environment.readEnv("BACKEND_CLIENT_AUTH_URL"));
-        var cognitoCredentials = new CognitoCredentials(backendClientCredentials::getId,
-                                                        backendClientCredentials::getSecret,
-                                                        cognitoServerUri);
-        var customerApiClient = new JavaHttpClientCustomerApiClient(httpClient, cognitoCredentials);
 
         var customer = fetchCustomerOrFailWithBadGateway(customerApiClient, customerAwareUserContext.customerUri());
 
         validatePublication(newPublication, customer);
+
+        setRightsRetention(newPublication, customer, customerAwareUserContext.username());
```

Comment:
I would try not to pass customer down the chain if possible. Rather just the RRS field.